infrastructure/tari_script/src/script.rs

```diff
@@ -368,7 +375,12 @@ impl TariScript {
         let target_height = stack.pop_into_number::<i64>()?;
         let block_height = i64::try_from(block_height)?;
 
-        let item = StackItem::Number(block_height - target_height);
+        // Here it is possible to underflow because the stack can take lower numbers where check
+        // height does not use a stack number and it's minimum can't be lower than 0.
+        let item = match target_height.checked_sub(block_height) {
```

Comment:
I believe the order is the wrong way around here when compared to the original? Why did it change?
```suggestion
        let item = match block_height.checked_sub(target_height) {
```

Reply:
The previous behaviour is incorrect based on the documentation: 

> /// Pops the top of the stack as `height`, then pushes the value of (`height` - the current height) to the stack.
    /// In other words, this opcode replaces the top of the stack with the difference between `height` and the
    /// current height. Fails with `InvalidInput` if there is not a valid integer value on top of the stack. Fails
    /// with `StackUnderflow` if the stack is empty.

So the question becomes which one is correct, the behaviour, or the documentation. I decided to trust the docs, and correct the behaviour. Do you have a particular reasoning for wanting to do the inverse?

Reply:
Spoke with @SWvheerden who agrees with you on this @hansieodendaal. So I've reverted the operands to the original positions and updated the op codes docs to match this behaviour.